### PR TITLE
Update gitignore

### PR DESCRIPTION
**Pull request by thaJeztah**
Adding some lines from the Moby gitignore

@dnephin @vdemeester for consideration; not sure if these would qualify as "should be present in your global ignore" as well

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,14 @@
+# if you want to ignore files created by your editor/tools,
+# please consider a global .gitignore https://help.github.com/articles/ignoring-files
+*.exe
+*.exe~
+*.orig
+.*.swp
 .DS_Store
+Thumbs.db
+# a .bashrc may be added to customize the build environment
+.bashrc
```
Comment (albers) [inline .gitignore]: The custom .bashrc only works in the context of the _moby/moby_ build scripts: https://github.com/moby/moby/blob/master/Dockerfile#L179. There's no corresponding feature in this repo.

Comment (thaJeztah) [reply .gitignore]: Ah, yes, makes sense, I can remove that, thanks 👍

**Comment by thaJeztah** [comment]
Updated, PTAL 😅

**Comment by thaJeztah** [comment]
To prevent things like https://github.com/docker/cli/pull/535

**Comment by vdemeester** [comment]
@dnephin @thaJeztah what is the current status of this ? 👼

**Comment by dnephin** [comment]
I think generally `.gitignore` should only include files that are created by a project, or are documented by the project, but should not be added to git. Other files should be ignored by the user.

I don't think adding these lines fixes the problem.

**Comment by vdemeester** [comment]
@dnephin although I agree, part of them can be created by the project (`*.exe`, the file ending with `~` or `*.swp` when editing with vim or else, …) and I really think it doesn't really hurt to add them…

… and the PR is stuck since September, it's creeping me out. So I'll go ahead and merge it and if we feel something wrong with it we can remove some of them 👼

**Pull request merged**